libConCoCt.py

```diff
@@ -19,29 +37,85 @@
 class Report(object):
     def __init__(self):
         self.parts = []
-    
+
     def add_part(self, report_part):
         self.parts.append(report_part)
-    
+
     def __str__(self):
         ret = ""
         for p in self.parts:
             ret += str(p)
         return ret
 
+    def to_json(self):
+        """
+        Converts data from this report to JSON format. First all data from
+        report parts and their messages are collected as dictionary. Then the
+        whole dictionary can be dumped to JSON and be returned.
+
+        TODO Check alternatives like using jsonpickle. Also it would be possible
+        to implement a Encoder class for the standard JSON module.
+        (See https://docs.python.org/2/library/json.html#json.JSONEncoder)
+        """
+        import json
+        json_object = {}
+        for part in self.parts:
+            json_object[part.source] = part.__to_json__()
+        return json.dumps(json_object)
+
+    def to_xml(self):
+        """
+        Builds a XML representation of all report parts in this report and all
+        messages inside these parts.
+
+        :returns: string containing XML representation of this report
+        """
+        # TODO Check if some serialization library like pyxser would be better.
+        report_root = xml.etree.ElementTree.Element('report')
+        for part in self.parts:
+            current_part = xml.etree.ElementTree.SubElement(report_root, part.source,
+                                                            attrib={'returncode': str(part.returncode)})
+            # create sub-element for each message in report part
+            for message in part.messages:
+                message_element = xml.etree.ElementTree.SubElement(current_part, 'message')
+                # get all information from message object (all fields that are not inherited by object class)
+                message_infos = filter( lambda x: x not in object.__dict__ , message.__dict__.keys() )
```

Comment:
This does not seem like a good idea. Maybe this could call a to_xml function in ReportPart/Message?